### DISCUSSION TITLE
Quarto Callout constructor

### DISCUSSION THREAD

**Discussion by Nenuial** [quarto-dev/quarto-cli#4442] [Q&A] [answered]
Hi,
I'm trying to write a shortcode that outputs a callout. It will take some information from the document meta and write the callout. I'm trying to use the new shortcode constructor (I'm using Quarto 1.3.211). Here's a simplified version of what I tried. The problem is, it doesn't show the callout. I'm obviously doing something wrong, but what?

In my document I have of course a `{{< doc-callout >}}` to call the shortcode.

```lua
return {
  ["doc-callout"] = function(args, kwargs, meta)
      local calloutHeader = pandoc.Header(2, "Test header")
      local calloutContent = pandoc.Inlines{"Some text ", pandoc.Emph("with part in italic")}
      
      local calloutDiv = {}
      calloutDiv["type"] = "note"
      calloutDiv["content"] = pandoc.Blocks{calloutHeader, calloutContent}
      
      calloutOut = quarto.Callout(calloutDiv)
      
      return calloutOut
  end
}
```

**Comment by cscheid** [comment, marked as answer]
You need to use the correct API: https://quarto.org/docs/prerelease/1.3/custom-ast-nodes/callout.html

**Reply by Nenuial** [reply]
I used that page. Isn't this what I do when I call `quarto.Callout()`?

Maybe I don't have the right parameters:
> a table with entries `type`, `caption`, and `condition`

What is `condition`? The caption is said to be optional in the list. But it says nothing about `condition`. I have `type` and `content`. Is something missing?

**Reply by cscheid** [reply]
Sorry, that's a typo. It should say `content`, and not `condition`.

Can you share a full example where you see this problem?

**Reply by Nenuial** [reply]
Here's a full extension with the code above:
https://github.com/Nenuial/doc-callout

**Reply by cscheid** [reply]
I now think this is just a bug in quarto.

The constructor works in a general Lua filter:

## qmd

```---
title: quarto.Callout test
filters:
  - test.lua
---

## A test

::: callout-replacement

Foo

:::
```

## test.lua

```lua
function Div(div)
  if div.classes:includes("callout-replacement") then
    return quarto.Callout({
        type = "note",
        caption = "Hello!",
        content = {pandoc.Para("This is a graf.")}
    })
  end
end
```

This produces this output:

<img width="536" alt="image" src="https://user-images.githubusercontent.com/285675/220388716-ee4b13ce-f36f-4e63-9219-5d633b1b7545.png">

But you're returning the constructor as the result of a _shortcode_. It's likely that this codepath is not working right.

**Reply by Nenuial** [reply]
Ok. Thanks for the answer. I should open an issue then?

**Reply by cscheid** [reply]
Ok, it's what I suspected.

Here's a quick workaround while I fix the bug: use `return pandoc.Plain({calloutOut})`.

**Reply by Nenuial** [reply]
Thanks a lot! I'll do that in the meantime.

**Reply by cscheid** [reply]
This should be fixed and available on https://github.com/quarto-dev/quarto-cli/releases/tag/v1.3.213

**Reply by Nenuial** [reply]
Thanks a lot! That was really quick. I can confirm it works.

**Reply by cscheid** [reply]
@Nenuial You might want to upgrade to a new release; there's a regression in Pandoc 3 that forced us to revert that feature, and v1.3.213 has the problem. You probably want https://github.com/quarto-dev/quarto-cli/releases/tag/v1.3.217

**Reply by Nenuial** [reply]
Thanks a lot for that info! I saw some strange errors because of the custom FiraSans font I use in lots of PDF documents. But I put it off as a problem for another day. Looks like it was because of that regression 😅
